src/ameba/rule/lint/literals_comparison.cr

```diff
@@ -11,39 +14,52 @@ module Ameba::Rule::Lint
   # YAML configuration example:
   #
   # ```
-  # Lint/StaticComparison:
+  # Lint/LiteralsComparison:
   #   Enabled: true
   # ```
-  class StaticComparison < Base
+  class LiteralsComparison < Base
     include AST::Util
 
     properties do
-      description "Identifies static comparisons"
+      description "Identifies comparisons between literals"
     end
 
     OP_NAMES = %w(=== == !=)
-    MSG      = "Comparison always evaluates to %s"
 
-    PRIMITIVES = {
+    MSG        = "Comparison always evaluates to %s"
+    MSG_LIKELY = "Comparison most likely evaluates to %s"
+
+    PRIMITIVE_LITERAL_TYPES = {
       Crystal::NilLiteral,
       Crystal::BoolLiteral,
       Crystal::NumberLiteral,
       Crystal::CharLiteral,
       Crystal::StringLiteral,
       Crystal::SymbolLiteral,
+      Crystal::ProcLiteral,
+    }
+
+    DYNAMIC_LITERAL_TYPES = {
       Crystal::RangeLiteral,
       Crystal::RegexLiteral,
       Crystal::TupleLiteral,
       Crystal::NamedTupleLiteral,
       Crystal::ArrayLiteral,
       Crystal::HashLiteral,
-      Crystal::ProcLiteral,
     }
 
+    LITERAL_TYPES =
+      PRIMITIVE_LITERAL_TYPES + DYNAMIC_LITERAL_TYPES
+
     def test(source, node : Crystal::Call)
       return unless node.name.in?(OP_NAMES)
       return unless (obj = node.obj) && (arg = node.args.first?)
-      return unless obj.class.in?(PRIMITIVES) && arg.class.in?(PRIMITIVES)
+
+      return unless obj.class.in?(LITERAL_TYPES) &&
+                    arg.class.in?(LITERAL_TYPES)
+
+      is_dynamic = obj.class.in?(DYNAMIC_LITERAL_TYPES) ||
+                   arg.class.in?(DYNAMIC_LITERAL_TYPES)
```

Comment:
This could check have a recursive check for the container types to see whether they consist of all primitives, and thus can be deemed _static_. I've left it as a potential further improvement, plus it's a pretty minor detail.